Firestore/src/Query.php

```diff
@@ -410,14 +387,121 @@ public function where($fieldPath, $operator, $value)
+                if (isset($nonCompositeFilter['unaryFilter'])) {
+                    unset($filters['fieldFilter']);
+                    $filters['unaryFilter'] = $nonCompositeFilter['unaryFilter'];
+                } else {
+                    $filters['fieldFilter'] = $nonCompositeFilter['fieldFilter'];
+                }
+            } elseif (isset($filter['fieldFilter'])) {
+                $filters[$key] = $this->findAndSetNonCompositeFilter($filter);
+            } elseif ($key === 'compositeFilter' && isset($filter['filters'])) {
+                $filters[$key]['filters'] = $this->createCompositeFilter($filter['filters']);
+            } elseif (isset($filter['compositeFilter']) &&
+            isset($filter['compositeFilter']['filters'])) {
+                $filters[$key]['compositeFilter']['filters'] =
+                    $this->createCompositeFilter($filter['compositeFilter']['filters']);
+            } else {
+                throw new \InvalidArgumentException(
+                    'The filter is not set correctly.'
+                );
+            }
+        }
+
+        return $filters;
+    }
+
+    /**
+     * Add a WHERE clause to the Query.
+     *
+     * For a list of all available operators, see
+     * {@see Google\Cloud\Firestore\V1\StructuredQuery\FieldFilter\Operator}.
+     * This method also supports a number of comparison operators which you will
+     * be familiar with, such as `=`, `>`, `<`, `<=` and `>=`. For array fields,
+     * the `array-contains`, `IN` and `array-contains-any` operators are also
+     * available.
+     * This method also supports usage of Filters (see {@see Google\Cloud\Firestore\Filter}).
+     * The Filter class helps to create complex queries using AND and OR operators.
+     *
+     * Example:
+     * ```
+     * $query = $query->where('firstName', '=', 'John');
+     * ```
+     *
+     * ```
+     * // Filtering against `null` and `NAN` is supported only with the equality operator.
+     * $query = $query->where('coolnessPercentage', '=', NAN);
+     * ```
+     *
+     * ```
+     * // Use `array-contains` to select documents where the array contains given elements.
+     * $query = $query->where('friends', 'array-contains', ['Steve', 'Sarah']);
+     * ```
+     *
+     * ```
+     * // Filtering with Filter::or condition
+     * $query = $query->where(Filter::or([Filter::condition('firstName', '=', 'John'),
+     *                          Filter::condition('firstName', '=', 'Monica')]));
+     * ```
+     *
+     * @param string|array|FieldPath $fieldPath The field to filter by, or array of Filters.
+     * @param string|int $operator The operator to filter by.
+     * @param mixed $value The value to compare to.
+     * @return Query A new instance of Query with the given changes applied.
+     * @throws \InvalidArgumentException If an invalid operator or value is encountered.
+     */
+    public function where($fieldPath, $operator = null, $value = null)
```

Comment:
rearrange the methods, public methods in a class should come above pre-existing private methods & the newly added ones.

Reply:
done!